raft/log.go

```diff
@@ -232,7 +232,7 @@ func (l *raftLog) term(i uint64) (uint64, error) {
 	if err == nil {
 		return t, nil
 	}
-	if err == ErrCompacted {
+	if err == ErrCompacted || err == ErrTermUnavailable {
```

Comment:
yes in etcdserver package in snapshot_merge.go for func createMergedSnapshotMessage. I have checked that code. In the code the err handling is already taken care.